static/app/components/events/viewHierarchy/wireframe.tsx

```diff
@@ -0,0 +1,146 @@
+import {useCallback, useEffect, useMemo, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+import {useResizeObserver} from '@react-aria/utils';
+
+import {ViewHierarchyWindow} from 'sentry/components/events/viewHierarchy';
+import {useMousePan} from 'sentry/components/events/viewHierarchy/useMousePan';
+
+type Rect = Pick<ViewHierarchyWindow, 'x' | 'y' | 'width' | 'height'>;
+
+export function getCoordinates(hierarchies: ViewHierarchyWindow[]) {
+  return hierarchies.map(hierarchy => {
+    const newHierarchy: Rect[] = [];
+
+    const nodesToProcess = [hierarchy];
+    while (nodesToProcess.length) {
+      const node = nodesToProcess.pop();
+
+      if (!node) {
+        continue;
+      }
+
+      // Fetch coordinates off the current node
+      newHierarchy.push({
+        x: node.x ?? 0,
+        y: node.y ?? 0,
+        width: node.width ?? 0,
+        height: node.height ?? 0,
+      });
+
+      if ((node.children ?? []).length !== 0) {
+        // Update the children's coords relative to the parent
+        const newChildren =
+          node.children?.map(child => ({
+            ...child,
+            x: child.x + (node.x ?? 0),
+            y: child.y + (node.y ?? 0),
```

Comment:
also not sure if we need to check for null/undefined here with `??` since `x` and `y` should always be defined, based on the type

Reply:
Okay, you're right because the type _says_ it should have x and y, but for now it looks like sometimes `x` and `y` aren't in the node. I'm going to follow this up with a PR to fix the type because it seems like some values that I'd expect to be there might not be.